### PR TITLE
♻️ 479 - create default fetch func for consent-api requests

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Replaces proxy route handling with an axiosClient and fetch util function that will access the Consent API directly. This is intended for use in server components only.

### Consent UI
- deletes `axiosClient` and renames `axiosProxyClient` to `consentApiClient`
- adds `consentApiFetch` (maybe can rename since it's axios, not fetch?) to be used as a base for all consent-api calls. uses `consentApiClient` and default config, and takes args for `url`, `headers`, `body`, `method`, and `session`. headers that are passed in can override default headers. The default config may need to be expanded.
- `consentApiFetch` will add the `accessToken` from the session to the Authorization header, if a session arg is passed in
- consent-api recaptcha middleware currently expects a recaptcha token in the body, so this can be passed in the `body` arg.
- *does not* yet remove proxy route handlers from `app/api` directory, but does replace the old axios client references with the new `consentApiClient`. Can discuss with the team if these route handlers may be useful if we need to do any client side api requests
- reorganizes the `services/api` folder

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/services/api/axios/consentApiClient.ts [hunk 1]
```diff
@@ -19,32 +19,35 @@
...
 import {
 	axiosErrorInterceptor,
 	axiosRequestInterceptor,
 	axiosResponseInterceptor,
-} from 'src/services/api/utils';
+} from './axiosInterceptors';
 
 const { CONSENT_API_URL, VERBOSE_AXIOS_LOGGING } = getAppConfig();
-const AXIOS_CLIENT_NAME = 'axiosProxyClient';
+const AXIOS_CLIENT_NAME = 'consentApiClient';
```
Comment (anncatton) [inline apps/consent-ui/src/services/api/axios/consentApiClient.ts:30]: renaming to indicate this axios client is dedicated to consent-api requests

File: apps/consent-ui/src/services/api/requests/status.ts [hunk 2]
```diff
@@ -19,14 +19,13 @@
...
+import { API } from 'src/constants';
+
+import consentApiFetch from '../axios/consentApiFetch';
 
 const getAPIStatus = async () => {
-	return await axiosClient
-		.get(urlJoin(PROXY_API_PATH, API.STATUS))
+	return consentApiFetch({ url: API.STATUS, method: 'GET' })
 		.then((res: AxiosResponse<APIStatus>) => {
 			if (res.status !== 200) {
```
Comment (anncatton) [inline apps/consent-ui/src/services/api/requests/status.ts:30]: Axios has a default `validateStatus` setting to resolve/reject based on the http status so this clause may not be needed https://axios-http.com/docs/req_config

File: apps/consent-ui/src/services/api/axios/consentApiFetch.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	},
+};
+
+const consentApiFetch = async ({
+	url,
+	method,
+	session,
+	body,
+}: {
+	url: string;
```
Comment (anncatton) [inline apps/consent-ui/src/services/api/axios/consentApiFetch.ts:39]: i should add the option to pass in headers as an arg as well

File: apps/consent-ui/src/services/api/axios/axiosInterceptors.ts
Comment (mistryrn) [inline apps/consent-ui/src/services/api/axios/axiosInterceptors.ts:1]: Ah, smart! 👌 💡

File: apps/consent-ui/src/services/api/requests/status.ts
Comment (mistryrn) [inline apps/consent-ui/src/services/api/requests/status.ts:1]: I like the re-org here 👏

File: apps/consent-ui/src/services/api/requests/status.ts (hunk 2 above)
Comment (mistryrn) [reply apps/consent-ui/src/services/api/requests/status.ts:30]: Interesting – did not know about that!

File: apps/consent-ui/src/services/api/axios/consentApiFetch.ts (hunk 3 above)
Comment (mistryrn) [reply apps/consent-ui/src/services/api/axios/consentApiFetch.ts:39]: Yeah that's a good idea 💡

**Review by mistryrn — COMMENTED** [review]
Looks great! Really like the new organization of the folders too.

Will be exciting to see how these work out and whether/not we'll need to apply `server-only` to anything 🙏

**Review by mistryrn — DISMISSED** [review]
Looks good to me 👍 Thanks for getting this sorted out!

File: apps/consent-ui/src/services/api/requests/status.ts (hunk 2 above)
Comment (samrichca) [reply apps/consent-ui/src/services/api/requests/status.ts:30]: if 200 is our only 2xx status code this is OK, if we're going to use 201 and 204 we can remove this

File: apps/consent-ui/src/services/api/requests/status.ts [hunk 4]
```diff
@@ -19,14 +19,13 @@
...
-import { API, PROXY_API_PATH } from 'src/constants';
-import { axiosClient } from 'src/services/api';
+import { API } from 'src/constants';
+
+import consentApiFetch from '../axios/consentApiFetch';
 
 const getAPIStatus = async () => {
-	return await axiosClient
-		.get(urlJoin(PROXY_API_PATH, API.STATUS))
+	return consentApiFetch({ url: API.STATUS, method: 'GET' })
```
Comment (samrichca) [inline apps/consent-ui/src/services/api/requests/status.ts:28]: what's the plan for POST requests?

File: apps/consent-ui/src/services/api/requests/status.ts (hunk 2 above)
Comment (anncatton) [reply apps/consent-ui/src/services/api/requests/status.ts:30]: we'll be using 201s on our `POST` request, so sounds like we can rely on the default config. i'll remove!

**Review by mistryrn — APPROVED** [review]
🚀

**Pull request merged**